### DISCUSSION TITLE
Why do ships have max speed?

### DISCUSSION THREAD

**Discussion by dmytrokyrychuk** [endless-sky/endless-sky#9305] [Ideas]
This is a bit of a silly question. I imagine that there is very little drag in vacuum. Engine thrust provides acceleration. What limits the ships max speed if an active engine can continue providing acceleration?

If we agree that the amount of drag in vacuum is sufficient to require engines to run continuously to maintain a certain speed, why don't the ships then decelerate and eventyally stop when the engines are turned off? Or do the decelerate, and I simply did not notice it before?

**Comment by warp-core** [comment]
Drag is a gameplay mechanic. It was never designed to be realistic. It was introduced because the a lack of some sort of max speed is not fun to play.
The deceleration is only applied while thrusting, if you stop thrusting, drag is not applied. Again, this isn't because it is designed to be realistic. Not needing to continue thrusting to maintain your velocity allows things like turning your ship to fire forward facing weapons at a pursuer. It also makes battery only builds usable. AI ships won't take advantage of this because they aren't programmed to.

**Comment by mOctave** [comment]
Also, there is a theoretical speed limit to anything in space. Obviously, you can't realistically go faster than light, but there's also the problem that as you approach significant fractions of light speed, very small particles can do a lot more damage. At a certain point, it _would_ be unsafe to keep accelerating, as space is only empty relative to Earth's atmosphere. One atom per cubic metre might not be much, but if that atom is going at relativistic velocities in relation to your ship, it's going to have a lot more mass and probably break something. If your ship is a 1m cube, and you're going at the speed of light, you're hitting about 300 million very heavy atoms per second, at very high speeds.

All this to say, that while drag is introduced as a gameplay measure, there are still actual restraints on how fast a theoretically super-powerful spacecraft could go.

**Reply by yafflehk** [reply]
That might be a fun mechanic though: you can go as fast as you like but beyond a certain speed your shields and hull will start to degrade, maybe engine damage as well?

**Reply by mOctave** [reply]
Fun dynamic, yes, but also a complete pain for new players... or for anyone used to "traditional" ES.

As a hazard zone, though, where damage scales with the square of velocity, it would be neat (but hard to code).
